### ISSUE TITLE
Augmentations in the model forward pass or in the dataloader?

### ISSUE THREAD

**Issue by ryanccarelli** [ryanccarelli/ssljax#33]
There isn't consensus in reference implementations about where to augment.

Lucidrains augments in (his equivalent to) the BaseSSL model forward pass
https://github.com/lucidrains/byol-pytorch/blob/master/byol_pytorch/byol_pytorch.py

DeepMind augments in the forward pass (in the loss function!)

Another natural thing to do is augment in the dataloader (augmentation happens right before passing data to Body)

We should benchmark different approaches
